### PR TITLE
Re-add `kinds` config field to tbot with a deprecation warning

### PR DESCRIPTION
**Pull request by timothyb89**
This re-adds the `kinds` config field with a deprecation warning. We removed the field in #11596 but due to strict YAML parsing this causes existing otherwise compatible configs to error out.

We should instead remove the field after a sane migration period in v11 or so.

### CODE REVIEW

**Review by strideynet — APPROVED** [review]

**Review by LKozlowski — APPROVED** [review]

File: tool/tbot/config/config_destination.go [hunk 1]
```diff
@@ -51,6 +51,10 @@ type DestinationConfig struct {
 	Roles   []string         `yaml:"roles,omitempty"`
 	Configs []TemplateConfig `yaml:"configs,omitempty"`
 
+	// Kinds is a deprecated field that remains for compatibility and should be
+	// removed in a future release in v11 or later.
```
[the remark this answers is not in the document]
Comment (zmb3) [reply tool/tbot/config/config_destination.go:55]: +1

File: tool/tbot/config/config_destination.go [hunk 2]
```diff
@@ -107,6 +111,12 @@ func (dc *DestinationConfig) CheckAndSetDefaults() error {
 		}
 	}
 
+	if len(dc.Kinds) > 0 {
+		log.Warnf("The `kinds` configuration field has been deprecated and " +
+			"will be removed in a future release. It is now a no-op and can " +
+			"safely be removed the configuration file.")
```
Comment (zmb3) [inline tool/tbot/config/config_destination.go:117]:
```suggestion
			"safely be removed from the configuration file.")
```

**Review by zmb3 — CHANGES_REQUESTED** [review]

File: tool/tbot/config/config_destination.go (hunk 2 above)
Comment (timothyb89) [reply tool/tbot/config/config_destination.go:117]: 🤦

File: tool/tbot/config/config_destination.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (timothyb89) [reply tool/tbot/config/config_destination.go]: Done!

**Review by zmb3 — APPROVED** [review]

**Pull request merged**